expr/plan.go

```diff
@@ -124,11 +157,11 @@ func consumeArg(args []*expr, j int, exp argType) (int, error) {
 // it also makes sure the kwarg has not been consumed already via the kwargs map
 // (it would be an error to provide an argument twice via the same keyword,
 // or once positionally and once via keyword)
-func consumeKwarg(optArgs []optArg, namedArgs map[string]*expr, k string, seenKwargs map[string]struct{}) error {
+func consumeKwarg(namedArgs map[string]*expr, k string, optArgs []arg, seenKwargs map[string]struct{}) error {
```

Comment:
move to expr.go or new args.go and call as `func (e *expr) ConsumeKwarg(key string, expected Arg) (int, error) {`

the whole seenKwargs should be handled by the caller.